server/worker/macos_setup_assistant.go

```diff
@@ -257,21 +272,29 @@ func (m *MacosSetupAssistant) runHostsTransferred(ctx context.Context, args maco
 			// after the cooldown period is over
 			level.Info(m.Log).Log("msg", "run hosts transferred: skipping assign profile for devices on cooldown", "serials", fmt.Sprintf("%s", skipSerials))
 		}
-		serials = assignSerials
-	}
 
-	if len(serials) == 0 {
-		level.Info(m.Log).Log("msg", "run hosts transferred: no devices to assign profile")
-		return nil
-	}
+		if len(assignSerials) == 0 {
+			level.Info(m.Log).Log("msg", "run hosts transferred: no devices to assign profile")
+			return nil
+		}
 
-	resp, err := m.DEPClient.AssignProfile(ctx, apple_mdm.DEPName, profUUID, serials...)
-	if err != nil {
-		return ctxerr.Wrap(ctx, err, "assign profile")
-	}
-	if err := m.Datastore.UpdateHostDEPAssignProfileResponses(ctx, resp); err != nil {
-		return ctxerr.Wrap(ctx, err, "worker: run hosts transferred")
+		for orgName, serials := range assignSerials {
+
+			decTok, err := assets.ABMToken(ctx, m.Datastore, orgName)
+			if err != nil {
+				return ctxerr.Wrap(ctx, err, "decrypting ABM token")
+			}
+			ctx = abmctx.NewContext(ctx, decTok)
+			resp, err := m.DEPClient.AssignProfile(ctx, orgName, profUUID, serials...)
+			if err != nil {
+				return ctxerr.Wrap(ctx, err, "assign profile")
+			}
+			if err := m.Datastore.UpdateHostDEPAssignProfileResponses(ctx, resp); err != nil {
+				return ctxerr.Wrap(ctx, err, "worker: run hosts transferred")
+			}
+		}
```

Comment:
sanity checking: this whole block used to run always, but now it will run only if `!fromCooldown`, is that intentional? I don't see why that needs to change but I'm probably missing something.

Reply:
Ah yes, I moved it in here because we need the org name -> serial mapping that comes from `ScreenDEPAssignProfileSerialsForCooldown`. However looking at it now, I think a better solution is to call `ScreenDEPAssignProfileSerialsForCooldown` outside the `if !fromCooldown`.